lib/api/LogManagerImpl.cpp

```diff
@@ -804,15 +820,37 @@ namespace MAT_NS_BEGIN
         return m_dataViewerCollection;
     }
 
-    void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
+    void LogManagerImpl::AddOrUpdateInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            m_dataInspectors.clear();
+            return;
+        }
+
+        auto itInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const auto& inspector){
+            return inspector->InspectorDescriptor().compare(dataInspector->InspectorDescriptor()) == 0;
+        });
+
+        if (itInspector != m_dataInspectors.end())
+        {
+            m_dataInspectors.erase(itInspector);
```

Comment:
Probably should add a `LOG_INFO` or something that we're replacing the dataInspector here. Or maybe add a boolean flag to denote if we want to replace if it exists.

Reply:
What would the situation be where we wouldn't want to update? When I'm thinking through it, in making a call to AddOrUpdate the desired behavior would be for the instance that you passed into that call to be present in that LogManager. I prefer the additional `LOG_INFO` to adding a boolean parameter